Waveform parameters

AdvancedImagingUTSW opened the pull request:
Meant for issue #413.

Ultimately, this is a partial PR. I want to check with @zacsimile before we go any further on how we want to implement certain aspects. As is, I do not think it should change operation so long as the smoothing percent value is at 0.

AdvancedImagingUTSW committed:
Widget callbacks

Will need to pass to the parent controller and the model next, as well as load the values from the configuration file.

AdvancedImagingUTSW committed:
Retrieve default values from configuration yaml file

Now loads default values from the configuration file.

Uses getter method should parameters not be listed.  Ideally, user should include a smoothing parameter in the remote_focus_device settings.

AdvancedImagingUTSW committed:
Some odd behavior.

Doesn't appear as if the smoothing parameter is working exactly as I hoped, at least after looking at the waveform plots.

Currently, when the waveform popup is called, it loads the values from the configuration.yaml file. However, given that this is a popup window, these values are killed if you close and open the window again.   Need to perhaps pull these values from the model.

AdvancedImagingUTSW committed:
Update waveform_popup_controller.py

Configuration now also updated...

Some of the naming in the waveform_popup_controller is confusing.

AdvancedImagingUTSW committed:
Spaces and variable definitiions.

Need to check and see if the waveform_tab_controller is pulling the same waveforms as the actual remote focusing device...

committing just as a good stopping spot.

AdvancedImagingUTSW committed:
Waveform smoothing

In waveform_popup_controller, refactor a dictionary and add some spacing for readability.

Fixed waveform smoothing by padding and truncating the waveform.

AdvancedImagingUTSW committed:
Update configuration.yaml and unit tests

Fix configuration file to account for smoothing.

Made test_waveforms naming consistent with other test files.

Added test to assert that the length of the smoothed waveform is the same as in the input waveform.

zacsimile reviewed (approved):
Appears to work on the multiscale.

AdvancedImagingUTSW committed:
Update remote_focus_base.py

annie-xd-wang committed:
add duty wait duration to waveform_constants.yml

annie-xd-wang committed:
Update waveform_parameter_popup_window.py

annie-xd-wang committed:
update waveform parameters

annie-xd-wang committed:
add duty wait duration time to the waveform

annie-xd-wang committed:
plot waveform if updated

annie-xd-wang committed:
Merge branch 'develop' into waveform_parameters

annie-xd-wang committed:
small fix

AdvancedImagingUTSW committed:
Variable name change.

Very minor change to make the variable more descriptive for what it actually does.
